graph-db/src/utils/bootstrapGremlinServer.js

```diff
@@ -0,0 +1,27 @@
+require('array-foreach-async')
+const { harvestCmrCollections } = require('./cmr/harvestCmrCollections')
+const { initializeGremlinConnection } = require('./gremlin/initializeGremlinConnection')
+const { indexCmrCollection } = require('./indexing/indexCmrCollection')
+
+/**
+ * Harvests collections from CMR and loads them into graph db
+ * @returns {Array} Errors from indexing into graph db
+ */
+exports.bootstrapGremilinServer = async () => {
+  const partitionedSearchResults = await harvestCmrCollections()
+  const gremlin = initializeGremlinConnection()
+  const indexingStatuses = []
+
+  for (const partition of partitionedSearchResults) {
+    console.log(`Indexing [${partition.length}] items into graph db`)
+    const indexingStatus = partition.map(
```

Comment:
I've not seen this construct  before, my gut tells me that in practice this loop won't actually wait but i could be entirely wrong -- I'd ensure the world (internet?) agrees with this.

Essentially having the async as an anonymous function inside of a loop, but the loop itself isn't awaiting.

Reply:
Ah, it looks like I forgot to add an `await` here and it should be `for await...of`

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for-await...of